### PR TITLE
Feat follow functionality

### PR DESCRIPTION
**Pull request by coldpigli**

### CODE REVIEW

File: src/components/UserCard/UserCard.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import { handleFollowUnfollow } from "services";
+
+const UserCard = ({currUserData}) => {
+
+    const {firstName, lastName, profileImg, username, _id} = currUserData;
+    const {authToken, userData} = useSelector((state)=>state.auth);
+    const dispatch = useDispatch();
+
+    const checkIfAlreadyFollowed = () => {
+        console.log("following", userData?.following);
```
Comment (nakulsharma15) [inline src/components/UserCard/UserCard.jsx:13]: You can remove console.log if it's no longer needed

**Review by nakulsharma15 — APPROVED** [review]
Looks good. Just a minor change. Nicely written code.

File: src/backend/db/posts.js [hunk 2]
```diff
@@ -16,13 +16,13 @@ export const posts = [
       likedBy: [],
       dislikedBy: [],
     },
-    username: "adarshbalika",
+    username: "arushii97",
     createdAt: formatDate(),
```
Comment (aishwarya-bm) [inline src/backend/db/posts.js:20]: Nice name😉

File: src/backend/db/posts.js [hunk 3]
```diff
@@ -74,4 +73,61 @@ export const posts = [
...
     updatedAt: formatDate(),
   },
+  {
+    _id: uuid(),
+    content:
+      "Hi, I'm Aaish and I'm new here. A little bit about me. You may see me as a single person but I am 4. Confusing right? Well no. I am the queen of cohorts. It's simple, I see a cohort, I join.",
```
Comment (aishwarya-bm) [inline src/backend/db/posts.js:79]: I see what you did there😂

File: src/utils/helperFunctions.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 export const checkUserPresence = (arr, value) => {
-    return arr.find((item)=>item.username === value)
+    return arr?.find((item)=>item.username === value)
 }
 
 export const checkIfBookmarked = (bookmarks, post) => {
-    return bookmarks.find((postId)=>postId === post._id);
+    return bookmarks.find((postId)=>postId === post?._id);
```
Comment (aishwarya-bm) [inline src/utils/helperFunctions.js:6]: Give optional chaining here too

**Review by aishwarya-bm — APPROVED** [review]
Everything looks good. Liked how consistently you are using Chakra utils and responsive capabilities. 
Remove console like logs, i see too many.

**Pull request merged**